### PR TITLE
Add better ES6 support

### LINKED ISSUE #774: ES6 Support
ES6 is about to be finalized and people are talking about it more frequently so it'll be sweet if highlight.js can support it.

### PR DESCRIPTION
**Pull request by Justineo**
to fix #774

- more keywords
- more built-ins
- new binary and octal literal formats
- template strings
- class definition

### CODE REVIEW

File: AUTHORS.en.txt [hunk 1]
```diff
@@ -159,3 +159,4 @@ Contributors:
 - Anthony Scemama <scemama@irsamc.ups-tlse.fr>
 - Taufik Nurrohman <latitudu.latitudu@gmail.com>
 - Pedro Oliveira <kanytu@gmail.com>
+- Gu Yiling <justice360@gmail.com>
```
Comment (ximyu) [inline AUTHORS.en.txt]: Yiling Gu?

Comment (Justineo) [inline AUTHORS.en.txt]: @ximyu I think it's ok to write in Chinese order.

**Comment by Sannis** [comment]
Thanks, merged!

**Comment by isagalaev** [comment]
JFYI: I cleaned up this code a bit more: https://github.com/isagalaev/highlight.js/compare/e8ea6263...7a8e337

**Comment by Sannis** [comment]
Thanks!
10 марта 2015 г. 22:32 пользователь "Ivan Sagalaev" <
notifications@github.com> написал:

**Comment by isagalaev** [comment]
… and I broke the build with it. Fixing now…

**Comment by mvpspl619** [comment]
When will this be released ?

**Comment by isagalaev** [comment]
We don't work against any schedule. Here's the progress towards the next release: https://github.com/isagalaev/highlight.js/milestones/8.5

**Comment by mvpspl619** [comment]
@isagalaev Gotcha. Thanks. Eagerly waiting for the 8.5 release.

**Pull request merged**